pkg/arvo/gen/start.hoon

```diff
@@ -3,32 +3,63 @@
 ::::  /hoon/start/gen
   ::
 :-  %say
-|=  *
+|=  [[now=@da * our=@p *] *]
 :-  %tang
 :~
 :+  %rose
 ["" "" ""]
-%+  turn
-  :~  "Welcome to Urbit.  This command-line interface is the Dojo.  You can use it to"
-      "control and operate your Urbit ship, or you can use the web interface if you"
-      "prefer (as long as this process is running).  That's probably at"
-      "http://localhost:8080 in Linux or http://localhost:80 in macOS."
-      "Try some commands out now:"
-      "  our"
-      "  now"
-      "You can run simple 'generators' (calculations) with a '+' like this:"
-      "  +vats"
-      "  +trouble"
-      "Or run a system-changing generator with a '|' like this:"
-      "  |hi ~zod"
+=/  =hart:eyre
+  .^  hart:eyre
+      %e
+      /(scot %p our)/host/(scot %da now)
+  ==
+=/  our-url
+  ;:  welp
+      "http"
+      ?:(p.hart "s" "")
+      "://"
+      (trip -.p.r.hart)
+      ?~(q.hart "" (welp ":" (a-co:co u.q.hart)))
```

Comment:
Arguably `our-url` shouldn't include the port number if `-.p.r.hart` is anything other than `'localhost'`.